table/block_based/block_based_table_builder.cc

```diff
@@ -1781,6 +1787,7 @@ void BlockBasedTableBuilder::WritePropertiesBlock(
     rep_->props.user_defined_timestamps_persisted =
         rep_->persist_user_defined_timestamps;
 
+    assert(IsEmpty() || rep_->props.key_largest_seqno != UINT64_MAX);
```

Comment:
Ah, good catch! I didn't notice it's legitimate to call `BlockBasedTableBuilder::Finish` right after its creation.

Reply:
In Flush we calls Abandon() if the output file is empty. We could probably do the same for compaction..